contrastscan-remote/src/main/java/com/contrastsecurity/scan/ContrastRemoteScanner.kt

```diff
@@ -0,0 +1,193 @@
+            //  case
+            throw IllegalArgumentException("Project $projectName is archived")
+        }
+        return project
+    }
+
+    private fun createClickableScanURL(projectId: String, scanId: String): URL {
+        val path = java.lang.String.join(
+            "/",
+            "",
+            "Contrast",
+            "static",
+            "ng",
+            "index.html#",
+            creds.orgId,
+            "scans",
+            projectId,
+            "scans",
+            scanId
+        )
+        return try {
+            URL(apiUrl.protocol, apiUrl.host, apiUrl.port, path)
+        } catch (e: MalformedURLException) {
+            throw RuntimeException(
+                "Error building clickable Scan URL. Please contact support@contrastsecurity.com for help", e)
+        }
+    }
+
+    private fun waitForResults(scan: Scan, scanOutputPath: Path, timeoutMs: Long) {
+        val scheduler = Executors.newSingleThreadScheduledExecutor()
+        try {
+            val reportsDirectory: Path = scanOutputPath.absolute().normalize().parent
+            try {
+                Files.createDirectories(reportsDirectory)
+            } catch (e: IOException) {
+                throw RuntimeException("Failed to create Contrast Scan reports directory", e)
+            }
+            val await = scan.await(scheduler)
+            val save = await.thenCompose { completed: Scan ->
+                CompletableFuture.runAsync(
+                    {
+                        try {
+                            completed.saveSarif(scanOutputPath)
+                        } catch (e: IOException) {
+                            throw UncheckedIOException(e)
+                        }
+                    },
+                    scheduler
+                )
+            }
+            val output = await.thenAccept { completed: Scan ->
+                val summary: ScanSummary
+                summary = try {
+                    completed.summary()
+                } catch (e: IOException) {
+                    throw UncheckedIOException(e)
+                }
+                writeSummaryToConsole(
+                    summary,
+                    { line: String? -> logger.debug(line) })
+            }
+            CompletableFuture.allOf(
+                save.toCompletableFuture(),
+                output.toCompletableFuture()
+            )[timeoutMs, TimeUnit.MILLISECONDS]
+        } catch (e: ExecutionException) {
+            // try to unwrap the extraneous ExecutionException
+            val cause = e.cause
+            // ExecutionException should always have a cause, but its constructor does not enforce this,
+            // so check if the cause is null
+            val inner = cause ?: e
+            throw RuntimeException("Failed to retrieve Contrast Scan results", inner)
+        } catch (e: InterruptedException) {
+            Thread.currentThread().interrupt()
```

Comment:
why this call?

Reply:
this was originally java code from our maven plugin that I copied and let the editor convert to Kotlin.
original code is: https://github.com/Contrast-Security-OSS/contrast-maven-plugin/blob/main/src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java#L304
I'll look more into the logic of this wait loop to see if there is a more idiomatic Kotlin-way to write it.